### PR TITLE
Address review comments on lesson 17

### LINKED ISSUE #185: Review comments on lesson 17
- [x] Ambiguous referent in "they are a shared resource" -- just say 
    that remote resources are shared.
- [x] "the scheduling system" is not a consequence.  Reword.
    (Also incorrect colon usage.)
- [x] Is "citizen" fraught?  Implies existence of resident aliens w/
    fewer rights?  "Community member" is maybe better? "Comrade"?
- [x] For head-node usage, quantitative limits on "quick tests" are bad, 
    the real issue is interference with other users.  Also, legit 
    uses include proving out workflows, debugging scripts.
- [x] Solution to the log-in node exercise should be clear that 
    running resource-intensive apps *on* *the* *login* *node*
    is what's bad, reinforces the context.
- [x] "...over lots of different input" should be plural.
- [x] In "backup plan", "secure copies" could imply "protected from 
    attack/surveillance", but in context we mean "protected
    from data loss due to device failure".  Can we use a different 
    adjective?
- [x] Latency is mentioned for the first time in the solution to the
    data-transfer exercise.  A natural place to introduce it 
    (and compare with bandwidth) is in the meta-data performance
    bullet of the preceding "transferring data" item.
- [x] Possibly a general "bandwidth vs latency" paragraph in the
    context of scaling would be useful?  Amdahl's law, even?

### PR DESCRIPTION
**Pull request by tkphd**
Closes #185.

### CODE REVIEW

File: _episodes/17-responsiblity.md [hunk 1]
```diff
@@ -25,32 +25,39 @@ one time varies from system to system but it is unlikely you will ever be the on
...
+compiling software. If the machine runs out of memory or processing capacity, it will become very
+slow and unusable for everyone. While the machine is meant to be used, be sure to do so responsibly
+&endash; in ways that will not adversely impact other users' experience.
 
-Don’t run jobs on the login node (though quick tests are generally fine). A “quick test” is
-generally anything that uses less than 5 minutes of time. If you use too much resource then other
-users on the login node will start to be affected - their login sessions will start to run slowly
-and may even freeze or hang.
+The login node is the right place to test out a workflow, e.g., making sure your script runs in the
+way you expect. It's also a good place to run code in a debugging environment, such as `gdb`,
```
[the remark this answers is not in the document]
Comment (tkphd) [reply _episodes/17-responsiblity.md:39]: Clusters are diverse, and opinions differ -- see #185.

File: _episodes/17-responsiblity.md [hunk 2]
```diff
@@ -62,17 +69,17 @@ with the *PID* to terminate any processes that are using large amounts of resour
...
+> > Running resource-intensive applications is frowned upon. Unless you are sure it will not affect
+> > other users, do not run jobs like #1 (`python`) or #4 (custom MD code). If you're unsure, ask
+> > your friendly sysadmin for advice.
 > {: .solution}
 {: .challenge}
 
 If you experience performance issues with a login node you should report it to the system
-staff (usually via the helpdesk) for them to investigate. You can use the `top` command
-to see which users are using which resources.
+staff (usually via the helpdesk) for them to investigate.
```
[the remark this answers is not in the document]
Comment (tkphd) [reply _episodes/17-responsiblity.md:82]: A full hour of unresponsive login node is way too long; a minute is too short. I'm comfortable leaving it to the judgment of the user. I think that by the time you *notice* a performance degradation, it's probably best to let the admins know.

File: _episodes/17-responsiblity.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (HaoZeke) [reply _episodes/17-responsiblity.md:39]: I'm still not sure there can be a use-case for debugging code on the login node. Surely that's something you can do on a workstation.

File: _episodes/17-responsiblity.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (HaoZeke) [reply _episodes/17-responsiblity.md:82]: Agreed.

File: _episodes/17-responsiblity.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (tkphd) [reply _episodes/17-responsiblity.md:39]: What if the cluster is a Power or Arm architecture, you don't have a workstation of that kind, and the memory error only seems to show up when you run on the cluster?

I tend to agree that running this as an interactive job on a node, but, I think it's heavy handed to *ban* such a use case. If the login node is idle, I would not object.

[the remark this answers is not in the document]
Comment (HaoZeke) [reply _episodes/17-responsiblity.md:39]: I think the exotic architecture and cross compilation case can be considered to be enough of a special case to not be part of the general guidelines.

I'd suggest either removing the line about debugging or at the very least amending it with a rider to ensure that we don't give people the wrong idea.

Also we are not / can not ban anything, we are trying to figure out a best practices guideline for users, not admins.

[the remark this answers is not in the document]
Comment (reid-a) [reply _episodes/17-responsiblity.md:39]: The intended scope (IMO) of the head-node debugging was primarily shaking out workflow issues -- unlike your laptop, the cluster may have modules that need to be loaded, possibly in a certain order, and paths or library versions that differ from your laptop, and doing an interactive test run on the head-node is a reliable, low-latency way to discover and fix these issues. Maybe "debugging" conveys the wrong idea, even if it's technically correct?

Some clusters may also have scarce, licensed commercial development tools, like the Intel or PGI compilers, or commercial parallel debugging tools, in which case users might be doing "real" development and debugging on the head node. I'm receptive to the argument that this is a niche case.

[the remark this answers is not in the document]
Comment (reid-a) [reply _episodes/17-responsiblity.md:39]: Suggested consensus text:

"The login node is a shared point of access for all cluster users, and should be treated respectfully. Cluster policies vary, but login nodes are always the right place to launch jobs, may be used for proving out workflows, and in some cases, may host advanced cluster-specific debugging or development tools."

[the remark this answers is not in the document]
Comment (tkphd) [reply _episodes/17-responsiblity.md:39]: Can we merge your last two comments, @reida? From Line 38, replace "The login node is the right place..." with

> Cluster policies vary, but login nodes are always the right place to launch jobs, may be used for proving out workflows, and in some cases, may host advanced cluster-specific debugging or development tools. The cluster may have modules that need to be loaded, possibly in a certain order, and paths or library versions that differ from your laptop, and doing an interactive test run on the head node is a quick and reliable way to discover and fix these issues.

[the remark this answers is not in the document]
Comment (reid-a) [reply _episodes/17-responsiblity.md:39]: Works for me, thanks.

[the remark this answers is not in the document]
Comment (tkphd) [reply _episodes/17-responsiblity.md]: 0c5bfa6

**Pull request merged**